### PR TITLE
Local Keycloak

### PR DESCRIPTION
**Pull request by borislavstoychev**
Please change ``` .env.local``` on frontend to

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -60,6 +60,29 @@ services:
...
+  keycloak:
+    container_name: keycloak
+    image: docker.io/bitnami/keycloak:15
+    environment:
+      - KEYCLOAK_USER=admin
+      - KEYCLOAK_PASSWORD=admin
+      - DEBUG=true
+      - DEBUG_PORT='*:8787'
+      - DB_VENDOR=POSTGRES
+      - DB_ADDR=pg-db
```
Comment (kachar) [inline docker-compose.yml:72]: I'm not sure it's a good idea to reuse the same db for backend and keycloak

Let's make another db instance dedicated to keycloak

File: docker-compose.yml [hunk 2]
```diff
@@ -60,6 +60,29 @@ services:
...
+      - TZ=Europe/Sofia
+    depends_on:
+      - pg-db
+    ports:
+      - '9080:8080'
+      - '8787:8787' # debug port
+    networks:
+      - backend-net
+
   # Flyway is deprecated, Prisma migrations are taking place
```
Comment (kachar) [inline docker-compose.yml:86]: We can remove all Flyway commended code in the same PR

File: docker-compose.yml [hunk 3]
```diff
@@ -60,6 +60,29 @@ services:
     networks:
       - backend-net
 
+  keycloak:
+    container_name: keycloak
+    image: docker.io/bitnami/keycloak:15
```
Comment (kachar) [inline docker-compose.yml:65]:
```suggestion
    # https://github.com/bitnami/bitnami-docker-keycloak
    image: docker.io/bitnami/keycloak:15
```

**Comment by kachar** [comment]
Let's consider adding our custom theme as described at https://github.com/bitnami/bitnami-docker-keycloak#adding-custom-themes

**Comment by borislavstoychev** [comment]
There is still the same 401 error unauthorized. I wonder if these warnings are the problem.
image

**Comment by Bunny303** [comment]
Did you add KEYCLOAK_SECRET in the api?

**Comment by borislavstoychev** [comment]
@Bunny303  you mean this secret key
image

**Comment by borislavstoychev** [comment]
image

**Comment by borislavstoychev** [comment]
@kachar  @Bunny303  the error was in my ```.env.local```  file on api. I had set KEYCLOAK_SECRET there and I probably never used this one in ```.env``` file.

**Comment by igoychev** [comment]
> Can we remove required email verification from dev realm? It will skip a step in user registration locally.

Yep, we can switch the verification off, and I was also thinking to import some default users for the main roles with the realm file, so that all comes ready-to-use. Will check how that can happen...

**Comment by kachar** [comment]
Great work!

Can we import multiple files?
If we can we might have one file that includes the real options and one file that includes the dev overrides and test users

**Comment by quantum-grit** [comment]
> Great work!
> 
> Can we import multiple files? If we can we might have one file that includes the real options and one file that includes the dev overrides and test users

Yes, multiple files per realm are possible, though I couldn't make it using the EXTRA-ARGS of the main keycloak container. It can be done with another container called keycloak-config-cli which allows realm import and updates from multiple files - see the commented section in docker-compose.yaml. 

Anyway, I did it with a single file without this new container to not load the dev machines, and we can split and extend easily when more complex config comes in place.

**Review by kachar — APPROVED** [review]
Marvelous work @borislavstoychev  and @igoychev :rocket: :rocket: :rocket:

**Comment by kachar** [comment]
> Can we remove required email verification from dev realm? It will skip a step in user registration locally.

It's done! Thanks for the suggestion @Bunny303

**Pull request merged**